Refactor: use .Site.GetPage for search input logic

Champbreed opened the pull request:
### Description
This PR refactors the search input logic in `layouts/partials/search-input.html` to use Hugo's dynamic `.Site.GetPage` method. 

### Related Issue
Fixes #50090

/sig-docs
/kind cleanup

lmktfy commented:
/area web-development

lmktfy commented on layouts/partials/search-input.html:
```diff
@@ -38,6 +38,6 @@
   >
 </div>
-{{ else if .Site.Params.customSearch -}}
+{{ else if .Site.GetPage "search" -}}
```
Why's this better? it's different logic, but I don't see why it's better logic.

Champbreed commented:
Hi @lmktfy,

I've updated the PR. I moved the logic to layouts/partials/search-input-custom.html and refactored it to use .Site.GetPage "search" as suggested. This replaces the manual path checking and allows Hugo to handle the search page resolution automatically.

Please let me know if this looks good to you now!

Champbreed commented:
Hi @lmktfy, I apologize for the previous broken state. I realized that my initial attempt removed the data-search-page attribute and the specific HTML structure that the site's JavaScript requires to initialize the search.

To fix this, I have:


- Restored the   < div class="search-bar" >  and the data-search-page attribute so the frontend logic can find the search index.

- Refactored the path logic to use {{- $searchPage := .Site.GetPage "/search" -}}, which is more robust than the manual file-path strings used previously.

I have verified the functionality in the latest Netlify deploy preview, and the search results are now appearing correctly. Thank you for your patience while I corrected the approach.

lmktfy commented on layouts/partials/search-input-custom.html:
```diff
@@ -1,19 +1,17 @@
+    {{ if $searchPage -}}
+    data-search-page="{{ "search/" | relLangURL }}"
+    {{- else -}}
     data-search-page="{{ "search/" | relURL }}"
```
I think we should use `.site.GetPage` here too. What is the original logic trying to achieve?

SayakMukhopadhyay replied:
@lmktfy I think this is fine. `$searchPage ` is `null` for languages that don't have a search template defined. So, for those languages, the `en` search page is used. 

But @Champbreed this can be made cleaner. See my reviews

Champbreed commented:
Apologies for the confusion, @lmktfy. Those were internal notes I accidentally left in the template during the refactor.
I now have a clean version with all comments removed. The logic remains exactly as discussed: using the $searchPage object to natively resolve the .RelPermalink. Ready for a final look!

SayakMukhopadhyay reviewed (changes requested):
Hello @Champbreed thank you for the PR. A major context to note is that we are trying to make our theme as close to Docsy as possible with the minimal number of overrides and even when we are overriding, we try to follow the Docsy way of doing things as much as possible. We want to be able to maintain the site easily with newer Docsy versions and hence this goal.

SayakMukhopadhyay reviewed (commented):
The code as it stands now is good to go but I have a few nits that would be great if they go in this PR itself, else we will put them in a new PR.

Champbreed commented:
@SayakMukhopadhyay I've updated search-input-custom.html to use the $searchLink variable and default function as suggested. This simplifies the template while maintaining Docsy parity. Ready for a final look!
